match_merger.py

```diff
@@ -11,22 +12,42 @@ class MatchMerger(Executor):
     The MatchMerger merges the results of shards by appending all matches..
     """
 
-    def __init__(self, default_traversal_paths: Tuple[str, ...] = ('r',), **kwargs):
+    def __init__(
+        self,
+        default_traversal_paths: Tuple[str, ...] = ('r',),
+        metric_name: str = 'cosine',
+        **kwargs
+    ):
         """
         :param default_traversal_paths: traverse path on docs, e.g. ['r'], ['c']
+        :param metric_name: metric_name to score matches.
         """
         super().__init__(**kwargs)
         self.default_traversal_paths = default_traversal_paths
+        self._metric_name = metric_name
 
     @requests
     def merge(self, docs_matrix: List[DocumentArray], parameters: dict, **kwargs):
+
+        metric_name = parameters.get('metric_name', self._metric_name)
+        top_k = int(parameters.get('top_k', 10))
         traversal_paths = parameters.get(
             'traversal_paths', self.default_traversal_paths
         )
+
         results = {}
         for docs in docs_matrix:
             self._merge_shard(results, docs, traversal_paths)
-        return DocumentArray(list(results.values()))
+        return self._select_top_k(
+            top_k, metric_name, DocumentArray(list(results.values()))
+        )
+
+    def _select_top_k(self, top_k, metric_name, docs):
```

Comment:
I would not have this as a separate function - it's used only once and barely saves any lines. Just put this in `merge`